src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -182,31 +188,103 @@ private StackFrame StackFrame(Chunk line)
 
             int lineNumber = line.ToInt32();
 
-            return string.IsNullOrEmpty(file)
-                ? LoadFrame(string.IsNullOrEmpty(function) ? line.ToString() : function, string.Empty, 0)
-                : LoadFrame(function, file, lineNumber);
+            if (string.IsNullOrEmpty(file))
+            {
+                return GetStackFrame(
+                    string.IsNullOrEmpty(function) ? line.ToString() : function,
+                    file: string.Empty,
+                    lineNumber: 0,
+                    fileProvider: _options.FileProvider,
+                    sourceCodeLineCount: _options.SourceCodeLineCount);
+            }
+            else
+            {
+                return GetStackFrame(function, file, lineNumber, _options.FileProvider, _options.SourceCodeLineCount);
+            }
         }
 
-        private StackFrame LoadFrame(string function, string file, int lineNumber)
+        internal static StackFrame GetStackFrame(
```

Comment:
No need for this to be static. Can just use `_fileProvider` inside it. This comment invalidates the other ones where you're using `_options.FileProvider`